src/mtcp/mtcp_restart.c

```diff
@@ -705,45 +771,22 @@ main(int argc, char *argv[], char **environ)
       //         mtcp_restart is statically linked, and doesn't need it.
       Area heap_area;
       MTCP_ASSERT(getMappedArea(&heap_area, "[heap]") == 1);
-      // FIXME:  choose higher of this
-      //           and lh_info.memRange.end; // 0x2aab00000000
-      start1 = heap_area.endAddr;
-      // FIXME:  choose lower of this and highMemStart
-      //        and make sure it doesn't intersect with lh_info.memRange
+      start1 = max(heap_area.endAddr, lh_info.memRange.end);
       Area stack_area;
       MTCP_ASSERT(getMappedArea(&stack_area, "[stack]") == 1);
-      end1 = stack_area.endAddr - 4 * GB;
-      // end1 = highMemStart - 4 * GB;
+      end1 = min(stack_area.endAddr - 4 * GB, highMemStart - 4 * GB);
       start2 = 0;
       end2 = start2;
```

Comment:
It looks like `start2` and `end2` are always 0? Then why do we need them?

Reply:
@jungan and @fyshhh, I'm not very sure either. But, I guess the intention was to show that if we want to block/reserve more than one region, then you can have another pair of `start` and `end`. This would be helpful if you want to reserve two small memory regions instead of one large region. Or if you have some restrictions in blocking one large region and you have to reserve in parts.
I believe @gc00  and Yao kept this code if at all they need it in the future but never used it.
**The question here is:** 
1. Do we need this code block or not?
2. Can we add a comment to describe the intention/purpose of `start2` and `end2`?

Reply:
Yeah, if would be helpful if we can have some comments here.

And If we really want to have some flexibility here, I think we can use a list of addresses.

Reply:
i think we can consider removing the entire block and maybe think of making it open for extension (like jun's suggestion). this is so that in the future, if we need three, or four, or more small regions, we don't just append them to the arguments. we can also use `stdarg` instead, although that might be less safe.

Reply:
Yes, I agree with removing these extra arguments. And, I'd suggest either changing it to a list or having a comment that currently, we reserve only one large area. If someone needs to extend it in the future then they can use it as a list (and give an example function signature).
@gc00 any comments or suggestions?